src/variant17.lib/variant17/Variant.h

```diff
@@ -59,40 +59,53 @@ auto selectType() {
 template<size_t N>
 using SelectType = UnwrapType<decltype(selectType<N>())>;
 
-/// Variant != std::variant
-/// * unchecked invalid state (only destruction is valid!)
-/// * simple recursive vistor
-/// * allows uncheck casts (you have to check before!)
-/// * sizeof(index) limits
 template<class... Ts>
-struct Variant {
+struct VariantWhich {
     static constexpr auto pack = to_type_pack<Ts...>;
     static constexpr auto indices = indexPackFor(pack);
-    using First = TypeHead<decltype(pack)>;
     using WhichValue = UnwrapType<SelectType<sizeof...(Ts)>>; // enough for npos!
-    enum { npos = sizeof...(Ts) }; // invalid state after exception - only destruction checks!
 
-    struct Which {
-        explicit constexpr Which(WhichValue v)
-            : value(v) {}
+    constexpr VariantWhich()
+        : value(0) {}
+    explicit constexpr VariantWhich(WhichValue v)
+        : value(v) {}
 
-        constexpr operator WhichValue() const { return value; }
+    constexpr operator WhichValue() const { return value; }
 
-        constexpr bool operator==(Which w) const { return w.value == value; }
-        constexpr bool operator!=(Which w) const { return w.value != value; }
+    constexpr bool operator==(VariantWhich w) const { return w.value == value; }
+    constexpr bool operator!=(VariantWhich w) const { return w.value != value; }
 
-        template<class T>
-        constexpr bool operator==(Type<T>) const {
-            return whichOf<T>() == *this;
-        }
-        template<class T>
-        constexpr bool operator!=(Type<T>) const {
-            return whichOf<T>() != *this;
-        }
+    template<class T>
+    constexpr bool operator==(Type<T>) const {
+        return whichOf<T>() == *this;
+    }
+    template<class T>
+    constexpr bool operator!=(Type<T>) const {
+        return whichOf<T>() != *this;
+    }
+
+    template<class T>
+    constexpr static auto whichOf(Type<T> = {}) -> VariantWhich {

```

Comment:
using this looks kind of stupid… `Which::whichOf` - it can simply be named `of` and be used as `Which::of`.